network/src/connectivity_manager/mod.rs

```diff
@@ -435,8 +436,14 @@ where
         // It enforces that a full node cannot have more outgoing connections than `connection_limit`
         // including in flight dials.
         let to_connect_size = if let Some(conn_limit) = self.outbound_connection_limit {
+            let outbound_connections = self
+                .connected
+                .iter()
+                .filter(|(_, metadata)| metadata.origin == ConnectionOrigin::Outbound)
```

Comment:
ahh, this is going to be interesting considering the PR I just reviewed, since this would effectively create some fun... so we need some form of this in that other PR such that the outbound is at least limited within the potential peerset ... that's some interesting configuration.

Reply:
> For testing, to test some changes via cluster test. This is not to be reviewed

The description was clear, it was for testing :)

Most of these changes are the other PR you reviewed, I was just testing what happens with 1, 2, or 3 failovers in a performance standpoint